services/core/MasterDriverAgent/tests/test_eagle.py

```diff
@@ -0,0 +1,309 @@
+        u'LinkStrength': u'0x64',
+        u'Channel': u'17'
+    }
+})
+
+command_instantaneous_demand = '<Command>\
+<Name>get_instantaneous_demand</Name>\
+<MacId>{}</MacId>\
+<Format>JSON</Format>\
+</Command>'.format(TEST_MAC_ID)
+
+response_instantaneous_demand = json.dumps({
+    u'InstantaneousDemand': {
+        u'DeviceMacId': u'0xffffffffffffffff',
+        u'Divisor': u'0x0000000a', # 10
+        u'TimeStamp': u'0x00000000',
+        u'MeterMacId': u'0xffffffffffffffff',
+        u'Multiplier': u'0x00000001', # 1
+        u'DigitsRight': u'0x00',
+        u'DigitsLeft': u'0x00',
+        u'SuppressLeadingZero': u'N',
+        u'Demand': u'0x000040' # 64
+    }
+})
+
+command_price_cluster = '<Command>\
+<Name>get_price</Name>\
+<MacId>{}</MacId>\
+<Format>JSON</Format>\
+</Command>'.format(TEST_MAC_ID)
+
+response_price_cluster = json.dumps({
+    u'PriceCluster': {
+        u'DeviceMacId': u'0xffffffffffffffff',
+        u'MeterMacId': u'0xffffffffffffffff',
+        u'Price': u'0x00000040', # 64
+        u'TrailingDigits': u'0x02', # 2
+        u'Tier': u'0',
+        u'Currency': u'0xffff',
+        u'StartTime': u'0xffffffff',
+        u'TimeStamp': u'0xffffffff',
+        u'Duration': u'0xffff'
+    }
+})
+
+command_current_summation = '<Command>\
+<Name>get_current_summation</Name>\
+<MacId>{}</MacId>\
+<Format>JSON</Format>\
+</Command>'.format(TEST_MAC_ID)
+
+response_current_summation = json.dumps({
+    u'CurrentSummation': {
+        u'DeviceMacId': u'0xffffffffffffffff',
+        u'Divisor': u'0x00000000',
+        u'TimeStamp': u'0x00000000',
+        u'MeterMacId': u'0xffffffffffffffff',
+        u'Multiplier': u'0x00000000',
+        u'DigitsRight': u'0x00',
+        u'DigitsLeft': u'0x00',
+        u'SuppressLeadingZero': u'N',
+        u'SummationReceived': u'0x00000000',
+        u'SummationDelivered': u'0x00000000'
+    }
+})
+
+command_demand_peaks = '<Command>\
+<Name>get_demand_peaks</Name>\
+<Format>JSON</Format>\
+</Command>'
+
+response_demand_peaks = json.dumps({
+    u'DemandPeaks': {
+        u'PeakReceived': u'-2.000000',
+        u'DeviceMacId': u'0xffffffffffffffff',
+        u'PeakDelivered': u'5.000000'
+    }
+})
+
+response_dict = {
+    command_network_status:       response_network_status,
+    command_instantaneous_demand: response_instantaneous_demand,
+    command_price_cluster:        response_price_cluster,
+    command_current_summation:    response_current_summation,
+    command_demand_peaks:         response_demand_peaks
+}
+
+
+def handle(env, start_response):
+    try:
+        length = env['CONTENT_LENGTH']
+        data = env['wsgi.input']
+        data = data.read(length)
+        data = data.replace(' ', '')
+    except KeyError:
+        data = None
+
+    start_response('200 OK', [('Content-Type', 'text/html')])
+    return response_dict[data]
+
+    return handle
+
+
+@pytest.fixture(scope='module')
+def agent(request, volttron_instance1):
```

Comment:
There is a new method that yield the results rather than using a finalizer so that you can just yield and then cleanup afterwards without having to do a separate function etc.  

I would like us to start moving toward this model as it seems to me as if it's cleaner code.

See https://github.com/VOLTTRON/volttron/blob/develop/services/core/VolttronCentral/tests/test_vc_discovery.py#L18 for an example.